plugin/evm/limitorders/memory_database.go

```diff
@@ -554,23 +562,38 @@ func (db *InMemoryDatabase) getTraderOrders(trader common.Address) []LimitOrder
 	traderOrders := []LimitOrder{}
 	for _, order := range db.OrderMap {
 		if strings.EqualFold(order.UserAddress, trader.String()) {
-			traderOrders = append(traderOrders, deepCopyOrder(*order))
+			traderOrders = append(traderOrders, deepCopyOrder(order))
 		}
 	}
 	return traderOrders
 }
 
-func (db *InMemoryDatabase) willReducePosition(order *LimitOrder) bool {
+func (db *InMemoryDatabase) getReduceOnlyOrderDisplay(order *LimitOrder) *LimitOrder {
 	trader := common.HexToAddress(order.UserAddress)
 	if db.TraderMap[trader] == nil {
-		return false
+		return nil
 	}
 	positions := db.TraderMap[trader].Positions
 	if position, ok := positions[order.Market]; ok {
 		// position.Size, order.BaseAssetQuantity need to be of opposite sign and abs(position.Size) >= abs(order.BaseAssetQuantity)
-		return position.Size.Sign() != order.BaseAssetQuantity.Sign() && big.NewInt(0).Abs(position.Size).Cmp(big.NewInt(0).Abs(order.BaseAssetQuantity)) != -1
+		if position.Size.Sign() == order.BaseAssetQuantity.Sign() {
+			return nil
+		}
+		if position.Size.CmpAbs(order.GetUnFilledBaseAssetQuantity()) == 1 {
+			// position is bigger than unfilled order size
+			orderCopy := deepCopyOrder(order)
+			return &orderCopy
+		} else {
+			// position is smaller than unfilled order
+			// display only the position size
+			orderCopy := deepCopyOrder(order)
+			orderCopy.BaseAssetQuantity = big.NewInt(0).Set(position.Size)
```

Comment:
modifying the baseAssetQuantity will generate a different orderHash and that is used to validate an order in the contracts, so that will fail.

Reply:
yep, overlooked it. Now I'm modifying the filled quantity so that the unfilled quantity is equal to position size